### DISCUSSION TITLE
1.44 - better DNS-mangling edition

### DISCUSSION THREAD

**Discussion by cerebrate** [arkane-systems/genie#191] [General]
* Standardized use of /usr/lib rather than /lib.
 * Updated ArkaneSystems.WSL to 0.2.13.
 * Made genie resolv.conf stub option-controlled.
 * Misc fixes.

Note that unlike 1.43, automatically adjusting /etc/resolv.conf for systemd-resolved use is an opt-in feature. See README for full details.

<hr /><em>This discussion was created from the release <a href='https://github.com/arkane-systems/genie/releases/tag/v1.44'>1.44 - better DNS-mangling edition</a>.</em>

**Comment by PavelSosin-320** [comment]
@cerebrate I'm not sure about other distros but in Fedora there is systemctl mask that is not persistent [Masking systemd units](https://fedoramagazine.org/systemd-masking-units/).  But in any scenario, if resolved is disabled applications that use internet services like repositories crash with famous "Temporary problem in name resolution", the same problem as "generate resolve.conf = yes" in the wsl.conf file. I think that the 1st step in Internet access logic involves access to resolved via D-Bus socket and if it fails the operation is aborted. Static /etc/resolved.conf doesn't work well for me and other people due to the heavy dependency of Win 11 from Azure cloud services. This decision is expected from the person who leads Microsoft today.
On other hand, the resolved.conf section
[Resolve]
Some examples of DNS servers which may be used for DNS= and FallbackDNS=:
Cloudflare: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001
Google:     8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
Quad9:      9.9.9.9 2620:fe::fe
**DNS= 192.168.1.1 9.9.9.9**
FallbackDNS=
Domains=
DNSSEC=true
DNSOverTLS=yes
MulticastDNS=no
LLMNR=resolve
Cache=yes
DNSStubListener=yes
ReadEtcHosts=yes
ResolveUnicastSingleLabel=no
is flexible enough to satisfy any network layout. I use implicitly DNSMasq forwarder in the 192.168.1.1. Any other static DNS address can be used keeping other options including ReadEtcHosts=yes. I don't see any limitation on using Bind, Unbind, etc.
